From 8e3c4b9597a3253d7d64805ba4da8444c577d632 Mon Sep 17 00:00:00 2001
From: Tony Camuso <tcamuso@redhat.com>
Date: Fri, 8 Aug 2014 18:14:36 -0400
Subject: [char] ipmi: info leak in compat_ipmi_ioctl()

Message-id: <1407521687-4858-3-git-send-email-tcamuso@redhat.com>
Patchwork-id: 87546
O-Subject: [RHEL7.1 PATCH 02/13] ipmi: info leak in compat_ipmi_ioctl()
Bugzilla: 1111214
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1111214

Cherry-picked without conflicts from the following upstream commit.

commit fa7df37b588f48a1ff6ef005187f3c5c2281df95
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Thu Sep 5 06:36:33 2013 -0500

    ipmi: info leak in compat_ipmi_ioctl()

    On x86_64 there is a 4 byte hole between ->recv_type and ->addr.

    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Corey Minyard <cminyard@mvista.com>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/char/ipmi/ipmi_devintf.c b/drivers/char/ipmi/ipmi_devintf.c
index d5a5f02..ec318bf 100644
--- a/drivers/char/ipmi/ipmi_devintf.c
+++ b/drivers/char/ipmi/ipmi_devintf.c
@@ -810,6 +810,7 @@ static long compat_ipmi_ioctl(struct file *filep, unsigned int cmd,
   struct ipmi_recv   __user *precv64;
   struct ipmi_recv   recv64;
 
+  memset(&recv64, 0, sizeof(recv64));
   if (get_compat_ipmi_recv(&recv64, compat_ptr(arg)))
    return -EFAULT;
 
-- 
1.7.1